WalletWasabi/WabiSabi/Client/CoinJoinTrackingDataFactory.cs

```diff
@@ -0,0 +1,36 @@
+using System.Collections.Generic;
+using System.Threading;
+using WalletWasabi.Blockchain.TransactionOutputs;
+using WalletWasabi.Wallets;
+using WalletWasabi.WebClients.Wasabi;
+
+namespace WalletWasabi.WabiSabi.Client;
+
+public class CoinJoinTrackingDataFactory
+{
+	public CoinJoinTrackingDataFactory(
+		IWasabiHttpClientFactory httpClientFactory,
+		RoundStateUpdater roundStatusUpdater,
+		CancellationToken cancellationToken)
+	{
+		HttpClientFactory = httpClientFactory;
+		RoundStatusUpdater = roundStatusUpdater;
+		CancellationToken = cancellationToken;
+	}
+
+	private IWasabiHttpClientFactory HttpClientFactory { get; }
+	private RoundStateUpdater RoundStatusUpdater { get; }
+	private CancellationToken CancellationToken { get; }
+
+	public CoinJoinTrackingData CreateCoinJoinTrackingData(Wallet wallet, IEnumerable<SmartCoin> coinCandidates)
```

Comment:
The `CoinJoinTrackingData` part is in the class name, so I think it's a bit extra. However, I think the important part is that we not only create here but also start something, so maybe:

```suggestion
	public CoinJoinTrackingData CreateAndStart(Wallet wallet, IEnumerable<SmartCoin> coinCandidates)
```

Reply:
ACK!